### PR TITLE
Fix No Case Property/Type Name Validation on Data Dictionary Imports

### PR DESCRIPTION
**Pull request by zandre-eng**
## Product Description

If the user tries to upload an Excel file that contains one or more invalid case property names, the following error will be shown:
Screenshot from 2024-01-03 12-51-43

A similar error as the one above will be shown if the user has an invalid case type name:
Screenshot from 2024-01-03 13-27-33

## Technical Summary

Link to ticket [here](https://dimagi-dev.atlassian.net/browse/SC-3253).

Additional validation has been added on case type/property names when importing an Excel file to the Data Dictionary. Currently, there already exists validation when creating case types/properties through the UI. This PR therefore aims to align the validation for both UI and Excel import processes.

## Feature Flag

None

### Safety story

- Local testing.
- Unit tests.

### Automated test coverage

An automated test has been created for the new `is_case_property_name_valid()` util function. Furthermore, the relevant unit test for testing import errors has been extended to account for this new error.

### QA Plan

No QA planned.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/data_dictionary/util.py [hunk 1]
```diff
@@ -360,3 +364,9 @@ def is_case_type_deprecated(domain, case_type):
         return case_type_obj.is_deprecated
     except CaseType.DoesNotExist:
         return False
+
+
+def is_case_type_or_prop_name_valid(case_prop_name):
```
Comment (SmittieC) [inline corehq/apps/data_dictionary/util.py:369]: I'm not sure where we currently do validation when creating a property/case type through the UI, but should we not use that rather?

Comment (zandre-eng) [reply corehq/apps/data_dictionary/util.py:369]: You can have a look [here](https://github.com/dimagi/commcare-hq/pull/33740) for how we will do the UI validation (PR still needs to be merged).

Given that the UI validation is done on the front-end through JavaScript and the import validation is being handled on the back-end through Django, we unfortunately wouldn't be able to easily use what we've done in the other PR. 

We could do the reverse of the above however, and instead move the front-end validation to the back-end to make use of the util function I've created, but I feel that refactor might be better suited for a separate PR.

Comment (SmittieC) [reply corehq/apps/data_dictionary/util.py:369]: Agreed, thanks for the clarification!

**Review by SmittieC — APPROVED** [review]
Looks good to me :+1:

**Comment by mkangia** [comment]
:+1:

**Pull request merged**